File: python/cucumber/model/catalog.py
import allure
import json
import requests

base_url = "https://api.restful-api.dev/objects"

headers = {
    'Content-Type': 'application/json',
    'accept': 'application/json'
}

def get(id):
    url = f"{base_url}/{id}"
    allure.attach(
        f"GET {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    result = requests.get(url=url, headers=headers)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result


def post(payload):
    url = f"{base_url}"
    allure.attach(
        f"POST {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    allure.attach(
        json.dumps(payload),
        name="Query payload",
        attachment_type=allure.attachment_type.JSON
    )
    result = requests.post(url=url,
                           data=json.dumps(payload),
                           headers=headers)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result


def delete(id):
    url = f"{base_url}/{id}"
    allure.attach(
        f"DELETE {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    result = requests.delete(url=url)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result

File: nodejs/cucumber/model/catalog.ts
import lodash from 'lodash';
import axios from 'axios';
import { World } from '@cucumber/cucumber';


const url = "https://api.restful-api.dev/objects";

const headers = {
  'Content-Type': 'application/json',
  'accept': 'application/json'
};


interface ICatalogItem {
  id: string;
  name: string;
  data: object;
  createdAt?: Date;
  [key: string]: any;
}


export class CatalogItem {

  id: string;
  name: string;
  data: object;
  createdAt: Date;

  constructor(obj: ICatalogItem) {
    this.id = obj['id'];
    this.name = obj['name'];
    this.data = lodash.cloneDeep(obj['data']);
    if ("createdAt" in obj)
      this.createdAt = obj['createdAt'];
  }
}


export class Catalog {

  world: World;

  constructor(world: World = null) {
    this.world = world;
  }

  async get(id: string) {
    if (this.world)
      this.world.attach(`GET ${url}/${id}`, "text/plain");
    return await axios.get(`${url}/${id}`);
  }

  async post(payload: object) {
    if (this.world) {
      this.world.attach(`POST ${url}`, "text/plain");
      this.world.attach("Payload:\n" + JSON.stringify(payload, null, 2));
    }
    return await axios.post(url, payload);
  }

  async delete(id: string) {
    if (this.world)
      this.world.attach(`DELETE ${url}/${id}`, "text/plain");
    return await axios.delete(`${url}/${id}`);
  }

}
